core/lib/basic_types/src/tee_types.rs

```diff
@@ -1,9 +1,58 @@
+use std::{fmt, str::FromStr};
+
 use serde::{Deserialize, Serialize};
-use strum::{Display, EnumString};
 
-#[derive(Debug, Clone, Copy, PartialEq, EnumString, Display, Serialize, Deserialize)]
+#[derive(Debug, Clone, Copy, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "lowercase")]
 #[non_exhaustive]
 pub enum TeeType {
-    #[strum(serialize = "sgx")]
     Sgx,
 }
+
+impl fmt::Display for TeeType {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            TeeType::Sgx => write!(f, "sgx"),
+        }
+    }
+}
```

Comment:
What do we need `Display` implementation for here? Display is [meant for user-facing output](https://doc.rust-lang.org/std/fmt/trait.Display.html), and I don't think we utilize this type for that purpose.
If it's used for _serialization_ (e.g. to store and then get data), then `serde` should be used instead.
For printing in logs, `Debug` is preferred.

If `Display` and `FromStr` are used to serialize and deserialize the type but not for anything else, I'd suggest removing this logic and rely on `serde` everywhere instead.

Reply:
As discussed offline, we need `Display` implementation in [`core/lib/dal/src/tee_proof_generation_dal.rs`](https://github.com/matter-labs/zksync-era/pull/2798/commits/ae107002f1d526df084b078cd7f66876eb9aefb0#diff-46d00b49fb68d6da13c9ecdc5ab3dab6adf502ec216349721586e96553652444).